### PR TITLE
Introduce CI

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

File: Readme.adoc [hunk 1]
```diff
@@ -57,9 +57,31 @@ You add the dependency to the Neo4j-Java-Driver driver in your `pom.xml`:
...
-To use the driver, you provide a connection URL, e.g. `bolt://user:password@localhost`, get a `Driver` instance, from which you get a Session.
+To use the driver, you provide a connection URL, e.g. `localhost`, and credentials, get a `Driver` instance, from which you get a Session.
 The session allows you to execute Cypher statements directly and iterate over the `StatementResults` to access the resulting `Value`s.
-Please edit example.movies.util.Util.java to add your connection settings, for instance "bolt://neo4j:neo4j@localhost"
+
+=== Configuration options
+
+[%header,cols=2*]
+|===
+|Environment variable name
```
Comment (jexp) [inline Readme.adoc:67]: possibly list the environment variable names here too.

File: Readme.adoc [hunk 2]
```diff
@@ -57,9 +57,31 @@ You add the dependency to the Neo4j-Java-Driver driver in your `pom.xml`:
...
+|Default value (or N/A)
+
+|PORT
+|8080
+
+|NEO4J_HOST
```
Comment (jexp) [inline Readme.adoc:73]: let's use NEO4J_URI with the full bolt URI

Comment (jexp) [reply Readme.adoc:73]: also /movies has no effect with the driver

File: Readme.adoc [hunk 3]
```diff
@@ -57,9 +57,31 @@ You add the dependency to the Neo4j-Java-Driver driver in your `pom.xml`:
...
+|demo.neo4jlabs.com/movies
+
+|NEO4J_USER
+|movies
+
+|NEO4J_PASS
```
Comment (jexp) [inline Readme.adoc:79]: NEO4J_PASSWORD for consistency (we should use the same env variables then everywhere)

File: src/main/java/example/movies/executor/CypherExecutor.java [hunk 4]
```diff
@@ -1,11 +1,55 @@
...
-    List<Map<String, Object>> query(String statement, Map<String,Object> params);
+public class CypherExecutor {
+
+    private final Driver driver;
+
+    private final String database;
+
+    public CypherExecutor(String url, String username, String password, String database) {
+        this.driver = GraphDatabase.driver(
+                String.format("neo4j://%s", url),
```
Comment (jexp) [inline src/main/java/example/movies/executor/CypherExecutor.java:25]: let's pass the full URI in here, so we don't construct it here.

File: src/main/java/example/movies/executor/CypherExecutor.java [hunk 5]
```diff
@@ -1,11 +1,55 @@
...
+                AuthTokens.basic(username, password),
+                Config.builder().withoutEncryption().build());
```
Comment (jexp) [inline src/main/java/example/movies/executor/CypherExecutor.java:27]: encryption is passed through the URI so no need to configure it here

File: src/main/java/example/movies/executor/CypherExecutor.java [hunk 6]
```diff
@@ -1,11 +1,55 @@
...
+        this.database = database;
+    }
+
+    public List<Map<String, Object>> query(String query, Map<String, Object> params) {
```
Comment (jexp) [inline src/main/java/example/movies/executor/CypherExecutor.java:31]: if we update this file, we can also simplify the whole project, which is now a bit overengineered :)

File: src/main/java/example/movies/Environment.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    public static int getPort() {
+        String webPort = System.getenv("PORT");
+        if (webPort == null || webPort.isEmpty()) {
+            return 8080;
+        }
+        return Integer.parseInt(webPort);
+    }
+
+    public static String getNeo4jUrl() {
+        String url = System.getenv("NEO4J_URL");
```
Comment (jexp) [inline src/main/java/example/movies/Environment.java:25]: NEO4J_URI

File: src/main/java/example/movies/Environment.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    public static String getNeo4jUsername() {
+        String user = System.getenv("NEO4J_USER");
+        if (user == null || user.isEmpty()) {
+            return DEFAULT_USER;
+        }
+        return user;
+    }
+
+    public static String getNeo4jPassword() {
+        String password = System.getenv("NEO4J_PASS");
```
Comment (jexp) [inline src/main/java/example/movies/Environment.java:49]: NEO4J_PASSWORD

File: src/main/java/example/movies/Environment.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
+package example.movies;
+
+/**
+ * @author Michael Hunger @since 22.10.13
+ */
+public class Environment {
+
+    private static final String DEFAULT_URL = "demo.neo4jlabs.com/movies";
```
Comment (jexp) [inline src/main/java/example/movies/Environment.java:8]: full URI `neo4j+s://demo.neo4jlabs.com`

**Review by jexp — CHANGES_REQUESTED** [review]
please see my comments

File: src/main/java/example/movies/executor/CypherExecutor.java (hunk 6 above)
Comment (fbiville) [reply src/main/java/example/movies/executor/CypherExecutor.java:30]: I would like to do that in a subsequent PR, once the CI is fully in place, if that's OK

Comment (fbiville) [reply src/main/java/example/movies/executor/CypherExecutor.java]: changed my mind, done.

**Review by jexp — APPROVED** [review]

**Pull request merged**